src/modules/deal/deals.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+import { DEFAULTS } from '@ukef/constants/properties';
+import { AcbsService } from '@ukef/modules/acbs/acbs.service';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+
+import { CreateDealDto } from './dto/deals-request.dto';
+
+@Injectable()
+export class DealsService {
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsService: AcbsService) {}
+
+  async createDeal(createDealDto: CreateDealDto): Promise<any> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
```

Comment:
Can we please wrap it in `try-catch`.

Reply:
Nest's default error handling will automatically convert any throwing of an internal exception within a controller to a 500 HTTP response ([docs](https://docs.nestjs.com/exception-filters#exception-filters-1)). These exceptions are all in `acbs-authentication.service.ts`. I don't think there is any need to catch them here, as that is already the behavior we want?